### PR TITLE
EmailCapture Component

### PR DESCRIPTION
**Pull request by Sahil-101**
### This PR: 
* Added EmailCapture, SurveySection, Interactive Section, Questions Components
* Made a front end for the issue #292 
* Currently the data is just logged into console, backend yet to be worked on.

### Self-Review Checklist
*(put x in between the brackets to mark them completed [x])*
- [x] PR title is clear with proper spelling and grammar
- [x] PR description contains a bulleted list of changes contained in the PR
- [x] PR links to relevant issues, with keywords to auto-close any issues fully resolved upon merge
- [x] All automated checks passed
- [x] Any variables introduced are named clearly and explicitly

### CODE REVIEW

File: components/email/EmailCaptureHelper.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+
+export default function EmailCaptureHome({ children, data }) {
```
Comment (LowLifeArcade) [inline components/email/EmailCaptureHelper.js:3]:
```suggestion
export default function EmailCaptureHelper({ children, data }) {
```

@audreyfeldroy says:
Either change EmailCaptureHome to match the name of the file, or rename the file to `EmailCaptureHome`.

File: components/email/InteractiveSection.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  const { cardsBackgroundColor } = useContextTheme();
+
+  const updateData = (childData) => {
+    setData((prevState) => {
+      return { ...prevState, ...childData };
+    });
+  };
+
+  const nextComponent = () => {
+    setComponentPointer(componentPointer + 1);
```
Comment (LowLifeArcade) [inline components/email/InteractiveSection.js:16]:
```suggestion
    setComponentPointer((componentPointer) => componentPointer + 1);
```

Let's do this to protect against stale data.

File: components/email/InteractiveSection.js [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  };
+
+  const previousComponent = () => {
+    setComponentPointer(componentPointer - 1);
```
Comment (LowLifeArcade) [inline components/email/InteractiveSection.js:20]: Same here :) 🌮

File: components/email/InteractiveSection.js [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  };
+
+  const childrenWithProps = React.Children.map(children, (child) => {
```
Comment (LowLifeArcade) [inline components/email/InteractiveSection.js:23]: @audreyfeldroy says:
Can we add a jsDoc comment to describe what is happening here?

**Review by LowLifeArcade — CHANGES_REQUESTED** [review]
Yo @Sahil-101! It's looking good. @audreyfeldroy had a few suggestions. After that we can merge this :)

File: components/email/InteractiveSection.js (hunk 4 above)
Comment (Sahil-101) [reply components/email/InteractiveSection.js:23]: Yeah Sure, but could you guide me doing that, I haven't done that anytime.

**Review by LowLifeArcade — APPROVED** [review]
Nice job @Sahil-101. Party time! 🍿 ✨

**Pull request merged**